runtime-sdk/src/error.rs

```diff
@@ -29,26 +29,171 @@ use crate::types::transaction::CallResult;
 /// ```
 pub trait Error: std::error::Error {
     /// Name of the module that emitted the error.
-    fn module_name() -> &'static str;
+    fn module_name(&self) -> &'static str;
 
     /// Error code uniquely identifying the error.
     fn code(&self) -> u32;
 
     /// Converts the error into a call result.
     fn to_call_result(&self) -> CallResult {
-        CallResult::Failed {
-            module: Self::module_name().to_owned(),
-            code: self.code(),
+        // If there is an underlying error, use that one.
```

Comment:
why is it more correct to discard the outer error?

Reply:
I'll change this so it will be up to the error definition to decide which one to use. See my reply about `transparent` above. Thoughts?

Reply:
ok moving discussion there